### PR TITLE
Restructure backup tests to include cluster re-creation

### PR DESCRIPTION
**Pull request by suleymanakbas91**
**Description**

Changes proposed in this pull request:

- Restructure backup tests to include cluster re-creation

**Related issue(s)**

See #4836

### CODE REVIEW

File: tests/end-to-end/backup-restore-test/deploy/chart/backup-test/values.yaml [hunk 1]
```diff
@@ -4,5 +4,5 @@ containerRegistry:
 nameOverride: backup-test
 
 image:
-  dir: develop/
-  tag: 164eb8ac
+  dir: pr/tests/backup-restore-test
```
Comment (Demonsthere) [inline tests/end-to-end/backup-restore-test/deploy/chart/backup-test/values.yaml]: This breaks the convention for images naming we have now. Maybe split it into `dir`, `name` and `tag`?

File: tests/end-to-end/backup-restore-test/backupe2e/service-catalog/helm_broker.go [hunk 2]
```diff
@@ -295,14 +276,6 @@ func (f *helmBrokerFlow) verifyDeploymentDoesNotContainRedisEnvs() error {
...
-	return f.deleteServiceBinding(redisBindingName)
-}
-
-func (f *helmBrokerFlow) deleteRedisInstance() error {
-	return f.deleteServiceInstance(redisInstanceName)
-}
-
 // that is necessary to restore correctly the etcd data
 // works only for single-member etcd cluster
 func (f *helmBrokerFlow) removeHelmBrokerEtcd() error {
```
Comment (polskikiel) [inline tests/end-to-end/backup-restore-test/backupe2e/service-catalog/helm_broker.go]: you can remove also that function

File: tests/end-to-end/backup-restore-test/backupe2e/service-catalog/helm_broker.go [hunk 3]
```diff
@@ -136,21 +132,6 @@ func (f *helmBrokerFlow) testResources() {
 	}
 }
 
-func (f *helmBrokerFlow) deleteResources() {
-	for _, fn := range []func() error{
-		f.deleteRedisBinding,
-		f.deleteRedisInstance,
-		f.verifyRedisInstanceRemoved,
```
Comment (polskikiel) [inline tests/end-to-end/backup-restore-test/backupe2e/service-catalog/helm_broker.go]: you can remove `f.verifyRedisInstanceRemoved` function definition

File: tests/end-to-end/backup-restore-test/backupe2e/service-catalog/brokers_flow.go [hunk 4]
```diff
@@ -187,16 +187,6 @@ func (f *brokersFlow) createBindingUsageAndWaitForReadiness(usageName, bindingNa
...
-	f.log.Infof("Deleting ServiceBinding %s", name)
-	return f.scInterface.ServicecatalogV1beta1().ServiceBindings(f.namespace).Delete(name, &metav1.DeleteOptions{})
-}
-
-func (f *brokersFlow) deleteServiceInstance(name string) error {
-	f.log.Infof("Deleting ServiceInstance %s", name)
-	return f.scInterface.ServicecatalogV1beta1().ServiceInstances(f.namespace).Delete(name, &metav1.DeleteOptions{})
-}
-
 func (f *brokersFlow) deleteBindingUsage(name string) error {
```
Comment (polskikiel) [inline tests/end-to-end/backup-restore-test/backupe2e/service-catalog/brokers_flow.go]: You can delete that function and its usages

File: tests/end-to-end/backup-restore-test/backupe2e/service-catalog/app_broker.go [hunk 5]
```diff
@@ -155,32 +151,10 @@ func (f *appBrokerFlow) testResources() {
 	}
 }
 
-func (f *appBrokerFlow) deleteResources() {
-	for _, fn := range []func() error{
-		f.deleteAPIServiceBindingUsage,
-		f.deleteAPIServiceBinding,
-		f.deleteEventsServiceInstance,
-		f.deleteAPIServiceInstance,
-		f.waitForInstanceDeleted,
```
Comment (polskikiel) [inline tests/end-to-end/backup-restore-test/backupe2e/service-catalog/app_broker.go]: You can delete `f.waitForInstanceDeleted` function definition

**Pull request merged**